lib/ansible/modules/cloud/cloudstack/cs_resourcelimit_facts.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016, René Moser <mail@renemoser.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: cs_resourcelimit_facts
+short_description: Get facts about resource limits on Apache CloudStack based clouds.
+description:
+    - Get facts about limits of resources for domains, accounts and projects.
+version_added: '2.9'
+author: Mauricio Accetturi Carvalho (@accetturi)
+options:
+  resource_type:
+    description:
+      - Type of the resource.
+    type: str
+    required: true
+    choices:
+      - instance
+      - ip_address
+      - volume
+      - snapshot
+      - template
+      - network
+      - vpc
+      - cpu
+      - memory
+      - primary_storage
+      - secondary_storage
+    aliases: [ type ]
+  domain:
+    description:
+      - Domain the resource is related to.
+    type: str
+  account:
+    description:
+      - Account the resource is related to.
+    type: str
+  project:
+    description:
+      - Name of the project the resource is related to.
+    type: str
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+- name: Retrieve a resource limit for instances of a domain
+  cs_resourcelimit_facts:
+    type: instance
+    domain: customers
+  delegate_to: localhost
+
+- name: Retrieve a resource limit for instances of an account
+  cs_resourcelimit_facts:
+    type: instance
+    account: moserre
+    domain: customers
+  delegate_to: localhost
+'''
+
+RETURN = '''
+---
+recource_type:
+  description: Type of the resource
+  returned: success
+  type: str
+  sample: instance
+limit:
+  description: Maximum number of the resource.
+  returned: success
+  type: int
+  sample: -1
+domain:
+  description: Domain the resource is related to.
+  returned: success
+  type: str
+  sample: example domain
+account:
+  description: Account the resource is related to.
+  returned: success
+  type: str
+  sample: example account
+project:
+  description: Project the resource is related to.
+  returned: success
+  type: str
+  sample: example project
+'''
+
+# import cloudstack common
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    cs_argument_spec,
+    cs_required_together,
+)
+
+RESOURCE_TYPES = {
```

Comment:
The CloudStack API accepts a resourcetypename parameter in the listResourceLimits endpoint. You could avoid the need to maintain this mapping if you used that instead of the resourcetype parameter. You'd just need to get the list from the docs, have them as choices in your module arguments and pass the chosen one directly.

Reply:
resourcetypename was added in 4.11 to the API, I would suggest to stick with this mapping for a bit longer to support cloudstack versions < 4.11